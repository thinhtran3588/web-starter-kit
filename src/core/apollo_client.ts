import ApolloClient, { gql } from 'apollo-boost';
import fetch from 'isomorphic-unfetch';
import { config } from '@app/config';

export const apolloClient = new ApolloClient({
  uri: config.apiEndpoint,
  fetch,
});
export { gql };
