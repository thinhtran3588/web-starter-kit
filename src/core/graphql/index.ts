export * from './queries/getCurrentUserQuery';
export * from './queries/getCurrentNotificationQuery';
export * from './apolloClient';
