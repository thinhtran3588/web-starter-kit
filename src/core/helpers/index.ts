export * from './format_with_lookup';
export * from './format_date';
export * from './format_date_time';
export * from './show_notification';
export * from './catch_error';
export * from './handle_error';
export * from './app_error';
export * from './sanitize_form_data';
export * from './get_error_message';
export * from './sleep';
