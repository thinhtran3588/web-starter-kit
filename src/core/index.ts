export * from './theme';
export * from './i18n';
export * from './interfaces';
export * from './helpers/format_with_lookup';
export * from './helpers/format_date';
export * from './helpers/format_date_time';
