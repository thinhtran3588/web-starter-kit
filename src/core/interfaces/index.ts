export * from './nav_item';
export * from './offset_pagination';
export * from './offset_pagination_result';
export * from './field_info';
export * from './filter';
export * from './filte_with_offset_pagination';
export * from './picker_data_item';
export * from './field_type';
export * from './field_value_type';
export * from './table_column';
export * from './next_request';
export * from './search_record';
export * from './user';
export * from './login_type';
export * from './form_data';
export * from './notification_content';
export * from './country';
export * from './error_with_code';
