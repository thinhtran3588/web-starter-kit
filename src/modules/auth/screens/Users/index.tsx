import React, { useState } from 'react';
import { WithTranslation } from 'react-i18next';
import debounce from 'lodash/fp/debounce';
import gql from 'graphql-tag';
import { useQuery } from '@apollo/react-hooks';
import { AdminLayout, FormHeader, FormSearch } from '@app/components';
import {
  FieldInfo,
  PickerDataItem,
  TableColumn,
  withTranslation,
  FilterWithOffsetPagination,
  OffsetPaginationResult,
  SearchRecord,
  formatWithLookup,
  formatDateTime,
} from '@app/core';
import { config } from '@app/config';
import { withApollo } from '@app/hoc';

interface Props extends WithTranslation {
  loginTypes: PickerDataItem<string>[];
  roles: PickerDataItem<string>[];
}

interface SearchResult {
  users: {
    data: SearchRecord[];
    pagination: OffsetPaginationResult;
  };
}

const GET_USERS_QUERY = gql`
  query getUsers($filter: String, $role: String, $loginType: String, $pageIndex: Int!, $itemsPerPage: Int!) {
    users(filter: $filter, role: $role, loginType: $loginType, pageIndex: $pageIndex, itemsPerPage: $itemsPerPage) {
      data {
        id
        email
        username
        fullName
        loginDetail {
          ... on FacebookLogin {
            loginType
          }
          ... on GoogleLogin {
            loginType
          }
          ... on EmailLogin {
            loginType
          }
          ... on PhoneNoLogin {
            loginType
          }
        }
        isActive
        registeredAt
        lastLoggedInAt
      }
      pagination {
        type
        total
      }
    }
  }
`;

const BaseUsers = (props: Props): JSX.Element => {
  const { t } = props;
  const loginTypes = [
    {
      value: '',
      label: t('all'),
    },
    ...props.loginTypes,
  ];
  const roles = [
    {
      value: '',
      label: t('all'),
    },
    ...props.roles,
  ];

  const filterFields: FieldInfo[] = [
    {
      name: 'filter',
      text: t('filter'),
    },
    {
      name: 'loginType',
      text: t('loginType'),
      type: 'picker',
      pickerDataSources: loginTypes,
    },
    {
      name: 'role',
      text: t('role'),
      type: 'picker',
      pickerDataSources: roles,
    },
  ];

  const columns: TableColumn[] = [
    {
      field: 'username',
      label: t('username'),
      minWidth: 100,
    },
    {
      field: 'fullName',
      label: t('fullName'),
      minWidth: 200,
    },
    {
      field: 'email',
      label: t('email'),
      minWidth: 120,
    },
    {
      field: 'phoneNo',
      label: t('phoneNo'),
      minWidth: 90,
    },
    {
      field: ['loginDetail', 'loginType'],
      label: t('loginType'),
      minWidth: 90,
      format: formatWithLookup(props.loginTypes),
    },
    {
      field: 'isActive',
      label: t('isActive'),
      minWidth: 90,
      align: 'center',
      format: (value) => (value ? t('active') : t('inactive')),
    },
    {
      field: 'lastLoggedInAt',
      label: t('lastLoggedInAt'),
      minWidth: 90,
      align: 'center',
      format: formatDateTime,
    },
    {
      field: 'registeredAt',
      label: t('registeredAt'),
      minWidth: 90,
      align: 'center',
      format: formatDateTime,
    },
  ];

  const createUser = async (): Promise<void> => {
    // TODO: implementation
  };

  const [filter, setFilter] = useState<FilterWithOffsetPagination>({
    pageIndex: 0,
    itemsPerPage: config.rowsPerPageOptions[0],
  });
  const setFilterDebounce = debounce(config.debounceDelay, setFilter);

  const onFilterChange = (newFilter: FilterWithOffsetPagination, useDebounce: boolean): void => {
    if (useDebounce) {
      setFilterDebounce(newFilter);
    } else {
      setFilter(newFilter);
    }
  };

  const { data } = useQuery<SearchResult>(GET_USERS_QUERY, {
    variables: filter,
  });

  return (
    <AdminLayout title='Users' description='Users'>
      <FormHeader
        breadcrumbLinks={[
          {
            text: 'Auth',
            link: '/admin/users',
          },
          {
            text: 'Users',
          },
        ]}
        commandButtons={[
          {
            text: 'Create',
            onClick: createUser,
            color: 'primary',
          },
        ]}
      />
      <FormSearch
        filterFields={filterFields}
        onFilterChange={onFilterChange}
        defaultFilter={{
          filter: '',
          loginType: loginTypes[0].value,
          role: roles[0].value,
        }}
        columns={columns}
        rows={data ? data.users.data : []}
        count={data ? data.users.pagination.total : 0}
      />
    </AdminLayout>
  );
};

BaseUsers.getInitialProps = async () => {
  const loginTypes = [
    {
      value: 'EMAIL',
      label: 'Email',
    },
    {
      value: 'PHONE_NO',
      label: 'Phone No',
    },
    {
      value: 'FACEBOOK',
      label: 'Facebook',
    },
    {
      value: 'GOOGLE',
      label: 'Google',
    },
  ];
  const roles = [
    {
      value: '0',
      label: 'Admin',
    },
    {
      value: '1',
      label: 'Normal User',
    },
  ];
  return {
    loginTypes,
    roles,
    namespacesRequired: ['common', 'admin_users'],
  };
};

export const Users = withApollo(withTranslation('admin_users')(BaseUsers));
