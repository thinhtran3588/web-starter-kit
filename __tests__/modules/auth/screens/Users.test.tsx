import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Users } from '@app/modules/auth/screens/Users';
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core';

describe('@app/modules/auth/screens/Users', () => {
  const providers = [
    {
      value: 'EMAIL',
      label: 'Email',
    },
    {
      value: 'PHONE_NO',
      label: 'Phone No',
    },
    {
      value: 'FACEBOOK',
      label: 'Facebook',
    },
    {
      value: 'GOOGLE',
      label: 'Google',
    },
  ];
  const theme = createMuiTheme({
    props: {
      MuiWithWidth: {
        initialWidth: 'xs',
      },
    },
  });
  beforeEach(() => {});

  it('renders successfully', async () => {
    const { baseElement } = render(
      <MuiThemeProvider theme={theme}>
        <Users providers={providers} />
      </MuiThemeProvider>,
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('navigates to Add Role page when clicking Create button', async () => {
    const { getByText } = render(
      <MuiThemeProvider theme={theme}>
        <Users providers={providers} />
      </MuiThemeProvider>,
    );

    fireEvent.click(getByText('Create'));
    // TODO: update test case
    expect(true).toBe(true);
  });
});
