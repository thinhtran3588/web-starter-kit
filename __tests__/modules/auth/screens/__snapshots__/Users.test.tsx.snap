// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@app/modules/auth/screens/Users renders successfully 1`] = `
<body>
  <div>
    <header
      class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed makeStyles-root-3 MuiAppBar-colorPrimary mui-fixed"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <next-link
          href="/"
        >
          <a
            href="/"
            title="Home"
          >
            <img
              alt="Logo"
              class="makeStyles-logo-5"
              src="/static/images/logo.svg"
            />
          </a>
        </next-link>
        <div
          class="makeStyles-flexGrow-4"
        />
        <div>
          <button
            aria-label="notification"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <div
                class="MuiAvatar-root makeStyles-icon-51 MuiAvatar-colorDefault"
              >
                EN
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <button
          aria-label="notification"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <span
              class="MuiBadge-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
                />
              </svg>
              <span
                class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary MuiBadge-dot"
              />
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="logout"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M21 3.01H3c-1.1 0-2 .9-2 2V9h2V4.99h18v14.03H3V15H1v4.01c0 1.1.9 1.98 2 1.98h18c1.1 0 2-.88 2-1.98v-14c0-1.11-.9-2-2-2zM11 16l4-4-4-4v3H1v2h10v3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          data-testid="menu-icon"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
    <div
      class="MuiDrawer-root MuiDrawer-docked"
      data-testid="sidebar"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation0 MuiDrawer-paper makeStyles-root-94 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft"
      >
        <ul
          class="MuiTreeView-root makeStyles-root-107"
          role="tree"
        >
          <li
            aria-expanded="true"
            class="MuiTreeItem-root makeStyles-rootItem-108 Mui-expanded makeStyles-expanded-111"
            role="treeitem"
            tabindex="-1"
          >
            <div
              class="MuiTreeItem-content makeStyles-content-109"
            >
              <div
                class="MuiTreeItem-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
              <div
                class="MuiTypography-root MuiTreeItem-label makeStyles-label-112 MuiTypography-body1"
              >
                <div
                  class="makeStyles-labelRoot-113"
                  style="color: rgb(26, 115, 232);"
                >
                  <span
                    class="makeStyles-labelIconContainer-114"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root makeStyles-labelIcon-115"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 1L3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm0 10.99h7c-.53 4.12-3.28 7.79-7 8.94V12H5V6.3l7-3.11v8.8z"
                      />
                    </svg>
                  </span>
                  <p
                    class="MuiTypography-root makeStyles-labelText-116 MuiTypography-body2"
                  >
                    Auth
                  </p>
                </div>
              </div>
            </div>
            <ul
              class="MuiCollapse-container MuiTreeItem-group makeStyles-group-110 MuiCollapse-entered"
              role="group"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper"
              >
                <div
                  class="MuiCollapse-wrapperInner"
                >
                  <li
                    class="MuiTreeItem-root makeStyles-rootItem-108"
                    role="treeitem"
                    tabindex="-1"
                  >
                    <div
                      class="MuiTreeItem-content makeStyles-content-109"
                    >
                      <div
                        class="MuiTreeItem-iconContainer"
                      >
                        <div
                          class="makeStyles-defaultEndIcon-118"
                        />
                      </div>
                      <div
                        class="MuiTypography-root MuiTreeItem-label makeStyles-label-112 MuiTypography-body1"
                      >
                        <next-link
                          href="/admin/users"
                        >
                          <a
                            class="makeStyles-link-117 makeStyles-labelRoot-113"
                            href="/admin/users"
                            title="Users"
                          >
                            <div
                              class="makeStyles-labelRoot-113"
                              style="color: rgb(26, 115, 232);"
                            >
                              <span
                                class="makeStyles-labelIconContainer-114"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root makeStyles-labelIcon-115"
                                  focusable="false"
                                  role="presentation"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                                  />
                                </svg>
                              </span>
                              <p
                                class="MuiTypography-root makeStyles-labelText-116 MuiTypography-body2"
                              >
                                Users
                              </p>
                            </div>
                          </a>
                        </next-link>
                      </div>
                    </div>
                  </li>
                  <li
                    class="MuiTreeItem-root makeStyles-rootItem-108"
                    role="treeitem"
                    tabindex="-1"
                  >
                    <div
                      class="MuiTreeItem-content makeStyles-content-109"
                    >
                      <div
                        class="MuiTreeItem-iconContainer"
                      >
                        <div
                          class="makeStyles-defaultEndIcon-118"
                        />
                      </div>
                      <div
                        class="MuiTypography-root MuiTreeItem-label makeStyles-label-112 MuiTypography-body1"
                      >
                        <next-link
                          href="/admin/roles"
                        >
                          <a
                            class="makeStyles-link-117 makeStyles-labelRoot-113"
                            href="/admin/roles"
                            title="Roles"
                          >
                            <div
                              class="makeStyles-labelRoot-113"
                              style="color: rgb(26, 115, 232);"
                            >
                              <span
                                class="makeStyles-labelIconContainer-114"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root makeStyles-labelIcon-115"
                                  focusable="false"
                                  role="presentation"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
                                  />
                                </svg>
                              </span>
                              <p
                                class="MuiTypography-root makeStyles-labelText-116 MuiTypography-body2"
                              >
                                Roles
                              </p>
                            </div>
                          </a>
                        </next-link>
                      </div>
                    </div>
                  </li>
                </div>
              </div>
            </ul>
          </li>
          <li
            aria-expanded="true"
            class="MuiTreeItem-root makeStyles-rootItem-108 Mui-expanded makeStyles-expanded-111"
            role="treeitem"
            tabindex="-1"
          >
            <div
              class="MuiTreeItem-content makeStyles-content-109"
            >
              <div
                class="MuiTreeItem-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
              <div
                class="MuiTypography-root MuiTreeItem-label makeStyles-label-112 MuiTypography-body1"
              >
                <div
                  class="makeStyles-labelRoot-113"
                  style="color: rgb(26, 115, 232);"
                >
                  <span
                    class="makeStyles-labelIconContainer-114"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root makeStyles-labelIcon-115"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
                      />
                    </svg>
                  </span>
                  <p
                    class="MuiTypography-root makeStyles-labelText-116 MuiTypography-body2"
                  >
                    Public Website
                  </p>
                </div>
              </div>
            </div>
            <ul
              class="MuiCollapse-container MuiTreeItem-group makeStyles-group-110 MuiCollapse-entered"
              role="group"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper"
              >
                <div
                  class="MuiCollapse-wrapperInner"
                >
                  <li
                    class="MuiTreeItem-root makeStyles-rootItem-108"
                    role="treeitem"
                    tabindex="-1"
                  >
                    <div
                      class="MuiTreeItem-content makeStyles-content-109"
                    >
                      <div
                        class="MuiTreeItem-iconContainer"
                      >
                        <div
                          class="makeStyles-defaultEndIcon-118"
                        />
                      </div>
                      <div
                        class="MuiTypography-root MuiTreeItem-label makeStyles-label-112 MuiTypography-body1"
                      >
                        <next-link
                          href="/admin/blogs"
                        >
                          <a
                            class="makeStyles-link-117 makeStyles-labelRoot-113"
                            href="/admin/blogs"
                            title="Blogs"
                          >
                            <div
                              class="makeStyles-labelRoot-113"
                              style="color: rgb(26, 115, 232);"
                            >
                              <span
                                class="makeStyles-labelIconContainer-114"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root makeStyles-labelIcon-115"
                                  focusable="false"
                                  role="presentation"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M21 5c-1.11-.35-2.33-.5-3.5-.5-1.95 0-4.05.4-5.5 1.5-1.45-1.1-3.55-1.5-5.5-1.5S2.45 4.9 1 6v14.65c0 .25.25.5.5.5.1 0 .15-.05.25-.05C3.1 20.45 5.05 20 6.5 20c1.95 0 4.05.4 5.5 1.5 1.35-.85 3.8-1.5 5.5-1.5 1.65 0 3.35.3 4.75 1.05.1.05.15.05.25.05.25 0 .5-.25.5-.5V6c-.6-.45-1.25-.75-2-1zm0 13.5c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5V8c1.35-.85 3.8-1.5 5.5-1.5 1.2 0 2.4.15 3.5.5v11.5z"
                                  />
                                  <path
                                    d="M17.5 10.5c.88 0 1.73.09 2.5.26V9.24c-.79-.15-1.64-.24-2.5-.24-1.7 0-3.24.29-4.5.83v1.66c1.13-.64 2.7-.99 4.5-.99zM13 12.49v1.66c1.13-.64 2.7-.99 4.5-.99.88 0 1.73.09 2.5.26V11.9c-.79-.15-1.64-.24-2.5-.24-1.7 0-3.24.3-4.5.83zM17.5 14.33c-1.7 0-3.24.29-4.5.83v1.66c1.13-.64 2.7-.99 4.5-.99.88 0 1.73.09 2.5.26v-1.52c-.79-.16-1.64-.24-2.5-.24z"
                                  />
                                </svg>
                              </span>
                              <p
                                class="MuiTypography-root makeStyles-labelText-116 MuiTypography-body2"
                              >
                                Blogs
                              </p>
                            </div>
                          </a>
                        </next-link>
                      </div>
                    </div>
                  </li>
                </div>
              </div>
            </ul>
          </li>
        </ul>
        <footer
          class="makeStyles-root-161"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            © 
            <next-link
              href="/"
            >
              <a
                href="/"
              >
                Thinh Tran
              </a>
            </next-link>
             
            2019
          </p>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            admin_common.createdWithLove
          </span>
        </footer>
      </div>
    </div>
    <div
      class="makeStyles-container-1"
    >
      <main
        class="makeStyles-main-2"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation1 makeStyles-root-162 MuiPaper-rounded"
        >
          <nav
            aria-label="breadcrumb"
            class="MuiTypography-root MuiBreadcrumbs-root makeStyles-breadcrumb-163 MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            <ol
              class="MuiBreadcrumbs-ol"
            >
              <li
                class="MuiBreadcrumbs-li"
              >
                <next-link
                  href="/admin/users"
                >
                  <a
                    href="/admin/users"
                  >
                    Auth
                  </a>
                </next-link>
              </li>
              <li
                aria-hidden="true"
                class="PrivateBreadcrumbSeparator-root-170 MuiBreadcrumbs-separator"
              >
                /
              </li>
              <li
                class="MuiBreadcrumbs-li"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
                >
                  Users
                </p>
              </li>
            </ol>
          </nav>
          <div
            class="makeStyles-commands-164"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary makeStyles-button-165"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Create
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation1 makeStyles-root-199 MuiPaper-rounded"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
            >
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-textField-304 MuiFormControl-marginDense MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense"
                  data-shrink="false"
                  for="filter"
                >
                  filter
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                    id="filter"
                    type=""
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
            />
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
            />
          </div>
          <div
            class="makeStyles-root-361 makeStyles-formTable-200"
          >
            <div
              class="makeStyles-tableWrapper-362"
              style="max-height: calc(100vh - 265px); min-height: 318px;"
            >
              <table
                class="MuiTable-root MuiTable-stickyHeader"
              >
                <thead
                  class="MuiTableHead-root"
                >
                  <tr
                    class="MuiTableRow-root MuiTableRow-head"
                  >
                    <th
                      class="MuiTableCell-root MuiTableCell-alignCenter MuiTableCell-head MuiTableCell-stickyHeader"
                      scope="col"
                      style="min-width: 100px;"
                    >
                      username
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-alignCenter MuiTableCell-head MuiTableCell-stickyHeader"
                      scope="col"
                      style="min-width: 200px;"
                    >
                      fullName
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-alignCenter MuiTableCell-head MuiTableCell-stickyHeader"
                      scope="col"
                      style="min-width: 120px;"
                    >
                      email
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-alignCenter MuiTableCell-head MuiTableCell-stickyHeader"
                      scope="col"
                      style="min-width: 90px;"
                    >
                      phoneNo
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-alignCenter MuiTableCell-head MuiTableCell-stickyHeader"
                      scope="col"
                      style="min-width: 90px;"
                    >
                      loginType
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-alignCenter MuiTableCell-head MuiTableCell-stickyHeader"
                      scope="col"
                      style="min-width: 90px;"
                    >
                      isActive
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-alignCenter MuiTableCell-head MuiTableCell-stickyHeader"
                      scope="col"
                      style="min-width: 90px;"
                    >
                      lastLoggedInAt
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-alignCenter MuiTableCell-head MuiTableCell-stickyHeader"
                      scope="col"
                      style="min-width: 90px;"
                    >
                      registeredAt
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="MuiTableBody-root"
                />
              </table>
            </div>
            <div
              class="MuiTablePagination-root"
            >
              <div
                class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
              >
                <div
                  class="MuiTablePagination-spacer"
                />
                <p
                  class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
                >
                  Rows per page:
                </p>
                <div
                  class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
                >
                  <div
                    aria-haspopup="listbox"
                    class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input MuiInputBase-inputSelect"
                    role="button"
                    tabindex="0"
                  >
                    10
                  </div>
                  <input
                    type="hidden"
                    value="10"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
                <p
                  class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
                >
                  0-0 of 0
                </p>
                <div
                  class="MuiTablePagination-actions"
                >
                  <button
                    aria-label="previous page"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                        />
                      </svg>
                    </span>
                  </button>
                  <button
                    aria-label="next page"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
  <div
    aria-hidden="true"
    id="simple-menu"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px; visibility: hidden;"
  >
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiMenu-paper MuiPaper-elevation8 MuiPopover-paper MuiPaper-rounded"
      role="document"
      style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
      tabindex="-1"
    >
      <ul
        class="MuiList-root MuiMenu-list MuiList-padding"
        role="menu"
        tabindex="-1"
      >
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          role="menuitem"
          tabindex="0"
        >
          English
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          role="menuitem"
          tabindex="-1"
        >
          Tiếng Việt
          <span
            class="MuiTouchRipple-root"
          />
        </li>
      </ul>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
